### PR TITLE
Feat: Add versioning for resume

### PR DESCRIPTION
**Pull request by tharshita**
# Notion Task

https://www.notion.so/Create-getLatestResume-api-a01c57da95c149e8a7be24835c9a5054

# Changes Made In This PR

- Edited resume saga to update version and prevResume upon open
- Replace all getResume calls with getLatestResume
- Versions panel renders list of versions ordered from latest to oldest

# UI Changes

Screenshot 2022-02-06 at 10 51 55 PM

### CODE REVIEW

File: src/components/builder/left/sections/Actions.js [hunk 1]
```diff
@@ -25,8 +25,9 @@ const Actions = ({ id }) => {
   const dispatch = useDispatch();
 
   const getSharableUrl = () => {
-    const shareId = resume.id;
-    return `https://b.careersocius.com/r/${shareId}`;
+    const resumeId = resume.id;
+    const { versionId } = resume;
+    return `https://b.careersocius.com/r/${resumeId}/v/${versionId}/`;
```
Comment (millasml) [inline src/components/builder/left/sections/Actions.js:30]: we shall always show latest version, so lets make the path only take in `resumeId`

File: src/components/builder/left/tabSections/VersionHistory.js [hunk 2]
```diff
@@ -47,15 +66,20 @@ const Versions = ({ id }) => {
               <Link
                 type="action"
                 underline="none"
-                value={getSharableUrl(version.versionNumber)}
-                onClick={handleOpenLink}
+                value={getSharableUrl(version.versionId)}
+                onClick={() => handleOpenLink(version.versionId)}
```
Comment (millasml) [inline src/components/builder/left/tabSections/VersionHistory.js:70]: lets make the panel not do anything for now!

File: src/pages/r.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ const ResumeRouter = () => (
   <Wrapper>
     <Router basepath="/r">
       <Redirect noThrow from="/" to="/" exact default />
-      <ResumeViewer path="/:id" />
+      <ResumeViewer path="/:id/v/:versionId" />
```
Comment (millasml) [inline src/pages/r.js:11]: only view latest resume, so dont need version id

File: src/pages/r/view.js [hunk 4]
```diff
@@ -48,7 +48,10 @@ const ResumeViewer = ({ id }) => {
       <div className={styles.container}>
         <Helmet>
           <title>{resume.name} | CareerSocius Resume Builder</title>
-          <link rel="canonical" href={`https://b.careersocius/r/${id}`} />
+          <link
+            rel="canonical"
+            href={`https://b.careersocius/r/${id}/v/${resume.versionId}`}
```
Comment (millasml) [inline src/pages/r/view.js:53]: only view latest resume, so dont need version id

**Review by millasml — CHANGES_REQUESTED** [review]
great job!! small comments :)

**Comment by tharshita** [comment]
okay done! :)

**Review by millasml — APPROVED** [review]

**Pull request merged**